Open Judge System/OJS.Services/Data/Participants/ParticipantsDataService.cs

```diff
@@ -51,24 +60,30 @@ public void Update(Participant participant)
                 .Select(x => x.IsOfficial)
                 .FirstOrDefault();
 
-        public void ChangeTimeForActiveInOnlineContestByContestIdAndMinutes(int contestId, int minutes)
-        {
-            var timeSpan = TimeSpan.FromMinutes(minutes);
-
-            var activeParticipants = this.participants
+        public IQueryable<Participant> GetOfficialInOnlineContestByCreatedOnAfterDateTimeAndBeforeDateTimeAndContest(
+            int contestId, 
+            DateTime after,
+            DateTime before) =>
+            this.participants
                 .All()
-                .Where(p => p.ContestId == contestId &&
+                .Where(p => 
+                    p.CreatedOn >= after &&
+                    p.CreatedOn <= before &&
+                    p.ContestId == contestId &&
                     p.IsOfficial &&
-                    p.Contest.Type == ContestType.OnlinePracticalExam &&
-                    p.ContestEndTime > DateTime.Now);
+                    p.Contest.Type == ContestType.OnlinePracticalExam);
 
-            foreach (var participant in activeParticipants)
-            {
-                participant.ContestEndTime = participant.ContestEndTime + timeSpan;
-                this.participants.Update(participant);
-            }
-
-            this.participants.SaveChanges();
-        }
+        public IQueryable<Participant> GetOfficialInOnlineContestByContestStartTimeAfterDateTimeAndBeforeDateTimeAndContest(
```

Comment:
Please, try finding a better name. I have been on the meetings where we decided what the functionality will be and yet I do not understand what the method does by its name...